cli/cmd/inject.go

```diff
@@ -122,6 +126,9 @@ func injectObjectMeta(t *metaV1.ObjectMeta, k8sLabels map[string]string, options
 	}
 	t.Annotations[k8s.CreatedByAnnotation] = k8s.CreatedByAnnotationValue()
 	t.Annotations[k8s.ProxyVersionAnnotation] = options.linkerdVersion
+	if options.noInitContainer {
+		t.Annotations[k8s.SetupIPTablesLabel] = strconv.FormatBool(options.noInitContainer)
+	}
```

Comment:
Ah, this relates to your recent comment:

> Right now, the implementation looks for the linkerd.io/setup-iptables annotation in order to prevent using the initContainer. This does mean that it isn't as "auto" as it could be. Is this okay? Do you have any other ideas to make the auto-inject without an initContainer more automatic?

I'm not sure we need this annotation. In the cni-plugin code, we could just check directly to see if the `linkerd-init` initContainer is present in the pod spec or not, without relying on a separate annotation. And in the proxy-injector code, this annotation will never be present, since the YAML that's sent to the proxy-injector has not previously been injected. I haven't reviewed the proxy-injector changes yet, but it seems like the proxy-injector should be using the value of its `-no-init-container` flag to make that determination.

Reply:
Yes, I agree. I will see what it looks like to not rely on the annotation.

Reply:
See the latest commits. I think that this is a good way to go. I will do some more testing to be sure that it works as it should, but so far it looks like it behaves similarly.

Reply:
Actually working through some issues now...the latest commit doesn't seem to work. Need to go back 2 commits for now.

Reply:
Never mind, I found the problem. I made a mistake in my cni-plugin code where true is always true and never false...imagine that. Will fix that tomorrow morning.